Decred Wallet/Features/App Launch/StartScreenViewController.swift

```diff
@@ -60,47 +60,56 @@ class StartScreenViewController: UIViewController {
     }
     
     func loadMainScreen() {
-        if !AppDelegate.walletLoader.isWalletCreated {
+        if AppDelegate.walletLoader.oneOrMoreWalletsExist {
+            self.checkStartupSecurityAndStartApp()
+        } else if DcrlibwalletWalletExistsAt(WalletLoader.appDataDir, BuildConfig.NetType) {
+            self.checkStartupSecurityAndLinkExistingWallet()
+        } else {
             self.displayWalletSetupScreen()
         }
-        else if StartupPinOrPassword.pinOrPasswordIsSet() {
-            self.promptForStartupPinOrPassword()
+    }
+    
+    func checkStartupSecurityAndStartApp() {
+        if StartupPinOrPassword.pinOrPasswordIsSet() {
+            self.promptForStartupPinOrPassword(completion: self.openWalletsAndStartApp)
         } else {
-            self.unlockWalletAndStartApp(pinOrPassword: "public") // unlock wallet using default public passphrase
+            self.openWalletsAndStartApp(startupPinOrPassword: "")
```

Comment:
The default passphrase should be "public". the app crash when you try to unlock an existing wallet

Reply:
Yes, the crash is an issue with dcrlibwallet that I've pushed a fix for. The default public passphrase for wallets is "public" whereas the startup passphrase has no default (or the default is ""). Previously, the startup passphrase was the same as the wallet's public passphrase but they've now been separated in dcrlibwallet.